packages/devtools_app/lib/src/landing_screen.dart

```diff
@@ -15,66 +16,111 @@ import 'theme.dart';
+    return Column(
+      children: [
+        _buildConnectDialog(),
+        const SizedBox(height: defaultSpacing),
+        _buildCodeSizeInstructions(),
+      ],
+    );
+  }
+
+  Widget _buildConnectDialog() {
+    return _buildSection(
+      title: 'Connect',
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          _headerText('Connect to a Running App'),
+          const SizedBox(height: denseRowSpacing),
+          _subHeaderText(
+              'Enter a URL to a running Dart or Flutter application'),
+          const Padding(padding: EdgeInsets.only(top: 20.0)),
+          _buildConnectInput(),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildCodeSizeInstructions() {
+    return _buildSection(
+      title: 'Code Size Tooling',
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          _headerText('Analyze and view diffs for your app\'s code size'),
+          const SizedBox(height: denseRowSpacing),
+          _subHeaderText('Load Dart AOT v8 snapshots, instruction sizes, or '
+              '"apk-analysis.json" files to track down code size issues'),
+          const Padding(padding: EdgeInsets.only(top: 20.0)),
+          RaisedButton(
+            child: const Text('Open code size tool'),
+            onPressed: () => Navigator.pushNamed(context, codeSizeRoute),
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildSection({@required String title, @required Widget child}) {
     final textTheme = Theme.of(context).textTheme;
     return Column(
-      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+      crossAxisAlignment: CrossAxisAlignment.start,
       children: [
-        Column(
-          crossAxisAlignment: CrossAxisAlignment.start,
-          children: [
-            Text(
-              'Connect',
-              style: textTheme.headline5,
-              key: const Key('Connect Title'),
-            ),
-            const PaddedDivider(),
-            Text(
-              'Connect to a Running App',
-              style: textTheme.bodyText1,
-            ),
-            const SizedBox(height: denseRowSpacing),
-            Text(
-              'Enter a URL to a running Dart or Flutter application',
-              style: textTheme.caption,
-            ),
-            const Padding(padding: EdgeInsets.only(top: 20.0)),
-            _buildTextInput(),
-            const PaddedDivider(padding: EdgeInsets.symmetric(vertical: 10.0)),
-            // TODO(https://github.com/flutter/devtools/issues/1111): support
-            // drag-and-drop of snapshot files here.
-          ],
+        Text(
+          title,
+          style: textTheme.headline5,
         ),
+        const PaddedDivider(),
+        child,
+        const PaddedDivider(padding: EdgeInsets.symmetric(vertical: 10.0)),
       ],
     );
   }
 
-  Widget _buildTextInput() {
+  Widget _headerText(String text) {
```

Comment:
nit: seems a little surprising that header is bodyText1 and subHeaderText is caption as the two don't seem that semantically related. Is there anything semantically similar in the material theme? It would be nice if these names generally matched names from the material theme.

Reply:
good point. fixed.